### ISSUE TITLE
Indian Local Server Unauthorized 403 

### ISSUE THREAD

**Issue by anuj-condenast** [cowinapi/developer.cowin#48]
I have a local server hosted by my college in India, I am trying to get a response using the public API but even that gives response 403. (It might not be geofencing)

Can anyone tell me how I can run from there?

P.S.: I am also trying to reach from an AWS EC2 instance but the same problem. 
@cowinapi

**Comment by regalstreak** [comment]
Pass a proper User-Agent header

**Comment by anuj-condenast** [comment]
@regalstreak Thanks for responding
I have tried in AWS using this
response = requests.get(request_api, headers = {"accept":"application/json", "Accept-Language": "hi_IN", "user-agent": "Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"})

Can you suggest something?

**Comment by regalstreak** [comment]
Have you tried setting the location to India? Foreign requests are blocked

**Comment by jhaanuj2108** [comment]
@regalstreak Do you mean while creating EC2 choosing the area as India or is there any other way ?

**Comment by regalstreak** [comment]
Yeah

**Comment by patelheggere** [comment]
https://cdn-api.co-vin.in/api/v2/auth/generateOTP

{
    "message": "Forbidden"
}
I am getting this error when I tried though post man

**Comment by mehimanshupatil** [comment]
@patelheggere 
​/v2​/auth​/generateOTP is protected api and not available for public
[https://apisetu.gov.in/public/marketplace/api/cowin/cowin-protected-v2#/](check here)

**Comment by shubhamtawde** [comment]
@mehimanshupatil I think it is a public API

**Comment by jhaanuj2108** [comment]
@shubhamtawde No it's private.
@patelheggere try this url
https://cdn-api.co-vin.in/api/v2/auth/public/generateOTP

**Comment by shubhamtawde** [comment]
@jhaanuj2108 I tried it , it comes under the public category. just now tried it, I got an OTP with the transaction ID as well. It's not private I guess.

**Comment by shubhamtawde** [comment]
@jhaanuj2108 btw do you know the format of confirm OTP endpoint?
{
  "otp": "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92",
  "txnId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
}
And does the protected API's use this token ahead? Thanks

**Comment by shubhamtawde** [comment]
@patelheggere don't directly open it in browser. You need to supply a request body with your mobile number. 
{
  "mobile": "1234567890"
}

**Comment by anuj-condenast** [comment]
> @jhaanuj2108 I tried it , it comes under the public category. just now tried it, I got an OTP with the transaction ID as well. It's not private I guess.

@shubhamtawde This one is private: https://cdn-api.co-vin.in/api/v2/auth/generateOTP
And this one is public: https://cdn-api.co-vin.in/api/v2/auth/public/generateOTP
Which one did you use? Notice that public has public after v2/auth

**Comment by anuj-condenast** [comment]
> @jhaanuj2108 btw do you know the format of confirm OTP endpoint?
> {
> "otp": "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92",
> "txnId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
> }

@shubhamtawde If you are using terminal then 
curl -X POST "https://cdn-api.co-vin.in/api/v2/auth/public/confirmOTP" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"otp\":\"8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92\",\"txnId\":\"3fa85f64-5717-4562-b3fc-2c963f66afa6\"}"

For endpoint URL I have tried various things like sending data payload, adding a header, appending in the URL itself but none of them seems to be working. If you find something then share it in this thread. Thanks


And regarding your other question:
> And does the protected API's use this token ahead? Thanks

No, the token generated from Public API's and also the token generated from Test Server in protected API can't be used in Production Server's protected APIs. I tried but didn't get any success.
